source/extensions/filters/network/mysql_proxy/mysql_message.cc

```diff
@@ -0,0 +1,142 @@
+#include "source/extensions/filters/network/mysql_proxy/mysql_message.h"
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/extensions/filters/network/mysql_proxy/mysql_codec.h"
+#include "source/extensions/filters/network/mysql_proxy/mysql_codec_clogin.h"
+#include "source/extensions/filters/network/mysql_proxy/mysql_codec_clogin_resp.h"
+#include "source/extensions/filters/network/mysql_proxy/mysql_codec_greeting.h"
+#include "source/extensions/filters/network/mysql_proxy/mysql_codec_switch_resp.h"
+#include "source/extensions/filters/network/mysql_proxy/mysql_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace MySQLProxy {
+
+ClientLogin MessageHelper::encodeClientLogin(AuthMethod auth_method, const std::string& username,
+                                             const std::string& password, const std::string& db,
+                                             const std::vector<uint8_t>& seed) {
+  ClientLogin login{};
+  login.setClientCap(CLIENT_SECURE_CONNECTION | CLIENT_LONG_PASSWORD | CLIENT_TRANSACTIONS |
+                     CLIENT_MULTI_STATEMENTS | CLIENT_PLUGIN_AUTH_LENENC_CLIENT_DATA);
+  if (db.length()) {
+    login.setClientCap(login.getClientCap() | CLIENT_CONNECT_WITH_DB);
+  }
+  login.setUsername(username);
+  login.setMaxPacket(0);
+  auto nseed = seed;
+  if (auth_method == AuthMethod::OldPassword) {
+    nseed.resize(8);
+    login.setAuthResp(OldPassword::signature(password, nseed));
+  } else {
+    nseed.resize(20);
+    login.setAuthResp(NativePassword::signature(password, nseed));
+    login.setClientCap(login.getClientCap() | CLIENT_PLUGIN_AUTH | CLIENT_PROTOCOL_41);
+    login.setAuthPluginName("mysql_native_password");
+  }
+  login.setDb(db);
+  login.setCharset(DEFAULT_MYSQL_CHARSET);
+  return login;
+}
+
+ClientLogin MessageHelper::encodeSslUpgrade() {
+  ClientLogin login{};
+  login.setClientCap(CLIENT_SSL);
+  login.setMaxPacket(0);
+  login.setCharset(DEFAULT_MYSQL_CHARSET);
+  return login;
+}
+
+ServerGreeting MessageHelper::encodeGreeting(const std::vector<uint8_t>& seed,
+                                             const std::string& auth_plugin_name) {
+  ServerGreeting greet{};
+  greet.setProtocol(MYSQL_PROTOCOL_10);
```

Comment:
Document that in this mode the assumed version of the server is 10.

Reply:
Ok, I'll add the doc after review.